themes/powerline-icon/powerline-icon.base.sh

```diff
@@ -0,0 +1,171 @@
+    elif [[ "${SCM_DIRTY}" -eq 1 ]]; then
+      color=${SCM_THEME_PROMPT_DIRTY_COLOR}
+    else
+      color=${SCM_THEME_PROMPT_CLEAN_COLOR}
+    fi
+    if [[ "${SCM_GIT_CHAR}" == "${SCM_CHAR}" ]]; then
+      scm_prompt+="${SCM_CHAR}${SCM_BRANCH}${SCM_STATE}"
+    fi
+    echo "${scm_prompt}${scm}|${color}"
+  fi
+}
+
+function __powerline_cwd_prompt {
+  echo "$(pwd | sed "s|^${HOME}|🏠|")|${CWD_THEME_PROMPT_COLOR}"
+}
+
+function __powerline_clock_prompt {
+  echo "$(date +"${THEME_CLOCK_FORMAT}")|${CLOCK_THEME_PROMPT_COLOR}"
+}
+
+function __powerline_battery_prompt {
+  local color=""
+  local battery_status="$(battery_percentage 2> /dev/null)"
+
+  if [[ -z "${battery_status}" ]] || [[ "${battery_status}" = "-1" ]] || [[ "${battery_status}" = "no" ]]; then
+    true
+  else
+    if [[ "$((10#${battery_status}))" -le 5 ]]; then
+      color="${BATTERY_STATUS_THEME_PROMPT_CRITICAL_COLOR}"
+    elif [[ "$((10#${battery_status}))" -le 25 ]]; then
+      color="${BATTERY_STATUS_THEME_PROMPT_LOW_COLOR}"
+    else
+      color="${BATTERY_STATUS_THEME_PROMPT_GOOD_COLOR}"
+    fi
+    ac_adapter_connected && battery_status="${BATTERY_AC_CHAR}${battery_status}"
+    echo "${battery_status}%|${color}"
+  fi
+}
+
+function __powerline_in_vim_prompt {
+  if [ -n "$VIMRUNTIME" ]; then
+    echo "${IN_VIM_THEME_PROMPT_TEXT}|${IN_VIM_THEME_PROMPT_COLOR}"
+  fi
+}
+
+function __powerline_left_segment {
+  local OLD_IFS="${IFS}"; IFS="|"
+  local params=( $1 )
+  IFS="${OLD_IFS}"
+  local separator_char="${POWERLINE_LEFT_SEPARATOR}"
+  local separator=""
+
+  if [[ "${SEGMENTS_AT_LEFT}" -gt 0 ]]; then
+    separator="$(set_color ${LAST_SEGMENT_COLOR} ${params[1]})${separator_char}${_omb_prompt_normal}"
+  fi
+  LEFT_PROMPT+="${separator}$(set_color - ${params[1]}) ${params[0]} ${_omb_prompt_normal}"
+  LAST_SEGMENT_COLOR=${params[1]}
+  (( SEGMENTS_AT_LEFT += 1 ))
+}
+
+function __powerline_last_status_prompt {
+  [[ "$1" -ne 0 ]] && echo "❌${1}|${LAST_STATUS_THEME_PROMPT_COLOR}" || echo "✅|${LAST_STATUS_THEME_PROMPT_COLOR_2}"
+}
+
+function __powerline_prompt_command {
+  local last_status="$?" ## always the first
+  local separator_char="${POWERLINE_PROMPT_CHAR}"
+
+  LEFT_PROMPT=""
+  SEGMENTS_AT_LEFT=0
+  LAST_SEGMENT_COLOR=""
+
+  # The IFS (internal field seperator) may have been changed outside to not contain
+  # the space character ' ' whence we need to make sure that the space separated list
+  # stored in POWERLINE_PROMPT is converted into an array correctly.
+  IFS=' ' read -r -a POWERLINE_PROMPT_ARRAY <<< "${POWERLINE_PROMPT}"
+
+  ## left prompt ##
+  for segment in ${POWERLINE_PROMPT_ARRAY[@]}; do
+    local info="$(__powerline_${segment}_prompt)"
+    [[ -n "${info}" ]] && __powerline_left_segment "${info}"
+  done
+  [[ "${last_status}" -ne 0 ]] && __powerline_left_segment $(__powerline_last_status_prompt ${last_status}) || __powerline_left_segment $(__powerline_last_status_prompt ${last_status})
```

Comment:
```suggestion
  __powerline_left_segment $(__powerline_last_status_prompt ${last_status})
```
The command executed when the condition `[[ "${last_status}" -ne 0 ]]` fails is exactly the same as the one the condition succeeds. These command shouldn't be repeated, but the command should be run without the condition.